dax/api.go

```diff
@@ -292,20 +293,50 @@ func (d *Dax) BatchGetItemPagesWithContext(aws.Context, *dynamodb.BatchGetItemIn
 	return d.unImpl()
 }
 
-func (d *Dax) QueryPages(*dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool) error {
-	return d.unImpl()
+func (d *Dax) QueryPages(input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool) error {
+	return d.QueryPagesWithContext(context.Background(), input, fn)
 }
 
-func (d *Dax) QueryPagesWithContext(aws.Context, *dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool, ...request.Option) error {
-	return d.unImpl()
+func (d *Dax) QueryPagesWithContext(ctx aws.Context, input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool, opts ...request.Option) error {
+	lastEvaluatedKey := input.ExclusiveStartKey
+
+	for {
+		out, err := d.QueryWithContext(ctx, input, opts...)
+		if err != nil {
+			return err
+		}
+
+		lastEvaluatedKey = out.LastEvaluatedKey
+		isLastPage := lastEvaluatedKey == nil
+		if !fn(out, isLastPage) {
+			break
+		}
+	}
+
+	return nil
 }
 
-func (d *Dax) ScanPages(*dynamodb.ScanInput, func(*dynamodb.ScanOutput, bool) bool) error {
-	return d.unImpl()
+func (d *Dax) ScanPages(input *dynamodb.ScanInput, fn func(*dynamodb.ScanOutput, bool) bool) error {
+	return d.ScanPagesWithContext(context.Background(), input, fn)
 }
 
-func (d *Dax) ScanPagesWithContext(aws.Context, *dynamodb.ScanInput, func(*dynamodb.ScanOutput, bool) bool, ...request.Option) error {
-	return d.unImpl()
+func (d *Dax) ScanPagesWithContext(ctx aws.Context, input *dynamodb.ScanInput, fn func(*dynamodb.ScanOutput, bool) bool, opts ...request.Option) error {
+	lastEvaluatedKey := input.ExclusiveStartKey
+
+	for {
+		out, err := d.ScanWithContext(ctx, input, opts...)
+		if err != nil {
+			return err
+		}
+
+		lastEvaluatedKey = out.LastEvaluatedKey
+		isLastPage := lastEvaluatedKey == nil
+		if !fn(out, isLastPage) {
+			break
+		}
```

Comment:
Current code has two infinite loop problems.

1. `input.ExclusiveStartKey` is not getting updated after each iteration and which leads to infinite calls to DAX with initial Key given by user.
2. User can make a mistake while defining the `fn` which will always results to `true` for all pages.   
 Eg: `fn :=func(page *dynamodb.QueryOutput, lastPage bool) bool { return true } ` can cause a infinite loop.

I would suggest to follow DynamoDB implementation. You can see it here https://github.com/aws/aws-sdk-go/blob/master/service/dynamodb/api.go#L3501